src/core/reference/src/op/concat.cpp

```diff
@@ -44,5 +44,33 @@ void concat(const std::vector<const char*>& args,
         }
     }
 }
+
+void concat(const std::vector<const std::string*>& args,
+            std::string* out,
+            const std::vector<Shape>& in_shapes,
+            const Shape& out_shape,
+            int64_t concatenation_axis,
+            size_t) {
+    size_t steps = 1;
+    for (int i = 0; i < concatenation_axis; ++i) {
+        steps *= out_shape[i];
+    }
+    const auto& shape_sizes = calculate_shape_sizes(in_shapes);
+
+    size_t out_offset = 0;
+    for (size_t step = 0; step < steps; ++step) {
```

Comment:
It looks like the main algo for this reference is duplicated. Is possible to make common function and inject only copy function depends on type which base on offset, size etc.?

Reply:
To be applied and reviewed within:
- https://github.com/openvinotoolkit/openvino/pull/23048